src/module/canvas/placeables/tokens/token-ruler.mjs

```diff
@@ -112,35 +112,59 @@ export default class DrawSteelTokenRuler extends foundry.canvas.placeables.token
   }
 
   /**
-   * @param {DeepReadonly<TokenRulerWaypoint>} waypoint
    * @inheritdoc
+   * @param {DeepReadonly<TokenRulerWaypoint>} waypoint
    */
   _getSegmentStyle(waypoint) {
-    const segment = super._getSegmentStyle(waypoint);
+    const style = super._getSegmentStyle(waypoint);
+
+    this._speedValueStyle(style, waypoint);
+
+    return style;
+  }
+
+  /**
+   * @inheritdoc
+   * @param {DeepReadonly<Omit<TokenRulerWaypoint, "index"|"center"|"size"|"ray">>} waypoint
+   * @param {DeepReadonly<foundry.grid.types.GridOffset3D>} offset
+   */
+  _getGridHighlightStyle(waypoint, offset) {
+    const style = super._getGridHighlightStyle(waypoint, offset);
+
+    this._speedValueStyle(style, waypoint);
+
+    return style;
+  }
+
+  /**
+   * Adjusts the grid or segment style based on the token's movement characteristics
+   * @param {{ color?: PIXI.ColorSource }} style        - The calculated style properties from the parent class
+   * @param {DeepReadonly<TokenRulerWaypoint>} waypoint - The waypoint being adjusted
+   * @protected
+   */
+  _speedValueStyle(style, waypoint) {
```

Comment:
Do we expext this to be subclassed? If not, I think we should default to private.